go/service/gregor.go

```diff
@@ -1506,11 +1503,23 @@ func (g *gregorHandler) DismissItem(cli gregor1.IncomingInterface, id gregor.Msg
 	if cli == nil {
 		cli = gregor1.IncomingClient{Cli: g.cli}
 	}
-	// TODO: Should the interface take a context from the caller?
-	err = cli.ConsumeMessage(context.TODO(), *dismissal)
+	err = cli.ConsumeMessage(ctx, *dismissal)
 	return err
 }
 
+// `cli` is the interface used to talk to gregor.
```

Comment:
Probably fails `go vet` because of comment.

Reply:
shockingly, it doesn't...